1/src/main.c

```diff
@@ -0,0 +1,102 @@
+// Link: https://adventofcode.com/2022/day/1
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#define MAX_INT_NUMBER_LENGTH 12
+#define TOP_LIST_SIZE 3
+
+void print_top_list(int* elf_calories, int* elf_calories_numbers, int size){
+    for(int i = size; i > 0; i--){
+        printf("place %d: calories -- %d, id: %d\n", size-i+1, elf_calories[i-1], elf_calories_numbers[i-1]);
+    }
+}
+void print_sum_of_elements(int arr[], int size) {
+    int sum = 0;
+    for (int i = 0; i < size; i++) {
+        sum += arr[i];
+    }
+    printf("total sum: %d\n", sum);
+}
+
+int find_insertion_index(int arr[], int size, int key) {
```

Comment:
```suggestion
unsigned int find_insertion_index(const Elf topElves[], unsigned int newCalories) {
```

Simplify the insertion logic by iterating through the list rather than using binary search, since our list size is small.